### DISCUSSION TITLE
How to modify theia default port

### DISCUSSION THREAD

**Discussion by pingbaohua** [eclipse-theia/theia#14094] [Q&A] [answered]
When I followed your instructions to set the environment variable THEIA-WEBVIEW.ETERNAL_ENDPOINT=192.168.2.134, it was indeed usable and the default port number was 443. However, when I specified a non default port number, such as 4455, when I started, the webview interface would not open. I tried to set THEIA-WEBVIEW.ETERNAL_ENDPOINT=192.168.2.134:4455, and a prompt appeared on the page: cannot get/webview/injx.html

**Comment by msujew** [comment, marked as answer]
Hey @pingbaohua,

You should specify `THEIA_WEBVIEW_EXTERNAL_ENDPOINT` to be literally the value `{{hostname}}` instead of any fixed IP or domain. It will later be replaced during the runtime of the application.

**Reply by pingbaohua** [reply]
thanks, solved this problem;
